### PR TITLE
lighter browser build

### LINKED ISSUE #37: Browser issues: size and setting headers
Hi again!
I've been playing around with reducing the size of the browser build. When using [browser-request](https://github.com/iriscouch/browser-request) and [statuses](https://github.com/jshttp/statuses), the size drops from ~1MB to ~40kB.

Building with browser-request is as simple as adding `browser: { 'request': 'browser-request'}` to package.json. Using statuses will replace `http.STATUS_CODES[code]` with `statuses[code]`, and the extra "burden" of another dependency. What do you think of a PR?

At the same time, I found out that [this commit](https://github.com/samcday/node-google-spreadsheets/commit/c86826374302fa17370653e4bca1f9d2ae7bb9da) broke the browser build (line 35 in spreadsheets.js). Setting headers in browser will force an "OPTIONS" request. Which fails, as [google does not set Access-Control-Allow-Origin on OPTIONS requests](http://stackoverflow.com/questions/28546969/cors-authorization-on-google-sheets-api-requests).

### PR DESCRIPTION
**Pull request by arve0**
- 1.5M -> 64k before uglify, 39k uglified
- use `browser-request` instead of `request`
- use `statuses` instead of node `http`
- fixes #37

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,16 +13,21 @@
...
     "semver": "^5.0.3"
   },
   "devDependencies": {
+    "browser-request": "^0.3.3",
     "browserify": "^12.0.1",
     "googleapis": "^2.1.6",
     "istanbul": "^0.4.0",
     "mocha": "^2.3.3",
     "should": "^7.1.1",
+    "statuses": "^1.2.1",
```
Comment (samcday) [inline package.json]: This should be in `dependencies` block, no?

Comment (arve0) [inline package.json]: Yes!

**Comment by arve0** [comment]
PS: This breaks the "google-spreadsheets fails gracefully on nonexistent spreadsheets"-test.

**Comment by samcday** [comment]
Bloody hell - sorry I let this one sit so long @arve0 :( Awesome stuff as usual! I fixed up the tests.

**Comment by samcday** [comment]
Pushed out `0.5.1` and updated dist in gh-pages branch.

**Comment by arve0** [comment]
No problem :smiley_cat:

**Pull request merged**